motionplan/plan.go

```diff
@@ -250,3 +251,50 @@ func (plan *SimplePlan) Path() Path {
 func (plan *SimplePlan) Trajectory() Trajectory {
 	return plan.traj
 }
+
+// NewExecutionState will construct an ExecutionState struct.
+func NewExecutionState(
+	plan Plan,
+	index int,
+	inputs map[string][]referenceframe.Input,
+	currentPose map[string]*referenceframe.PoseInFrame,
```

Comment:
it is possible for any of these arguments to be nil when passed into the function?
If so, we should add a nilness check.

Reply:
I agree.